Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java

```diff
@@ -22,42 +22,303 @@
 
+import com.microsoft.azure.storage.blob.CloudBlobClient;
+import com.microsoft.azure.storage.blob.CloudBlobContainer;
+import com.microsoft.azure.storage.blob.CloudBlockBlob;
+import com.microsoft.azuretools.azurecommons.helpers.NotNull;
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+
 import javax.swing.*;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
 import java.lang.reflect.Constructor;
 import java.lang.reflect.Method;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.Map;
-
+import java.net.URI;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.logging.Logger;
+/**
+ * Use JxBrowser to load URL for Swing, AWT or JavaFX UIs
+ */
 public class JxBrowserUtil {
+    public static final Logger log = Logger.getLogger(JxBrowserUtil.class.getName());
+    public static int MAX_RETRY_TIMES = 3;
+    public static int MAX_WAIT_TIME_IN_SECONDS = 120;
+
+    /** Mapping OS to package name */
+    private static final Map<String, String> platformToJxBrowserJar = Collections.unmodifiableMap(
+            new HashMap<String, String>() {{
+                put("win32", "jxbrowser-win32-6.16.jar");
+                put("win64", "jxbrowser-win32-6.16.jar");
+                put("mac", "jxbrowser-mac-6.16.jar");
+                put("linux64", "jxbrowser-linux64-6.16.jar");
+                put("linux32", "jxbrowser-linux64-6.16.jar");
+            }}
+        );
+
+    private static final String JXBROWSER_CLASS_BROWSERVIEW = "com.teamdev.jxbrowser.chromium.swing.BrowserView";
+    private static final String JXBROWSER_CLASS_BROWSER = "com.teamdev.jxbrowser.chromium.Browser";
+    private static final String AZURE_ACCOUNT_NAME = "ltiantest";
+    private static final String AZURE_BLOB_URI = "http://ltiantest.blob.core.windows.net";
+    private static final String JXBROWSER_LICENSE_FILE = "license.jar";
+    private static final String JXBROWSER_COMMON_JAR = "jxbrowser-6.16.jar";
+    private static final int RETRY_BACKOFF_FACTOR = 3;
+
+    /**
+     * Async version of createBrowserViewAndLoadURL, return FutureTask<JComponent>
+     * @param url
+     * @return
+     */
+    public static FutureTask<JComponent> createBrowserViewAndLoadURLAsync(@NotNull final String url, final String targetPath) {
+        FutureTask<JComponent> createAndLoadTask = new FutureTask<> (new Callable<JComponent>() {
+            @Override
+            public JComponent call() throws JxBrowserException {
+                return createBrowserViewAndLoadURL(url, targetPath);
+            }
+        });
+
+        Thread thread = new Thread(createAndLoadTask);
+        thread.setDaemon(true);
+        thread.start();
```

Comment:
the thread should be closed when we close IntelliJ or Eclipse?

Reply:
Yes, it is daemon thread and will exit if the whole process exit.

---
In reply to: [156552639](https://github.com/Microsoft/azure-tools-for-java/pull/1296#discussion_r156552639) [](ancestors = 156552639)